vmm/src/lib.rs

```diff
@@ -1020,16 +1020,18 @@ impl Vmm {
             memory_model::GuestMemoryError::MemoryNotInitialized,
         ))?;
 
-        // The vcpu_count has a default value. We shouldn't have gotten to this point without
-        // having set the vcpu count.
-        let vcpu_count = self.vm_config.vcpu_count.ok_or(VcpusNotConfigured)?;
-
         #[cfg(target_arch = "x86_64")]
         arch::x86_64::configure_system(
             vm_memory,
             GuestAddress(arch::x86_64::layout::CMDLINE_START),
             kernel_config.cmdline.len() + 1,
-            vcpu_count,
+            // If the length of the vcpus cannot be turned into an `u8`,
```

Comment:
The vcpu count is validated in machine_config.rs/validate_vcpu_num  against MAX_SUPPORTED_VCPUS and I am not sure if we actually need to validate again vs 256.

Reply:
You could just pass the vcpus.len() as u8 or into().

Reply:
> The vcpu count is validated in machine_config.rs/validate_vcpu_num against MAX_SUPPORTED_VCPUS and I am not sure if we actually need to validate again vs u8 max value.

Here I am actually using the MAX_SUPPORTED_VCPUS as an argument for why the expect is safe. I am not trying to re-validate against MAX_SUPPORTED_VCPUS here.

Reply:
> You could just pass the vcpus.len() as u8 or into().

Indeed `try_into` complicates things a bit. I used `as u8`. Take a look.